test/inference/Inference.jl

```diff
@@ -12,13 +12,25 @@ include(dir*"/test/test_utils/AllUtils.jl")
     if VERSION > v"1.2"
         @testset "threaded sampling" begin
             # Test that chains with the same seed will sample identically.
-            chain1 = sample(Random.seed!(5), gdemo_default, HMC(0.1, 7), MCMCThreads(),
-                            1000, 4)
-            chain2 = sample(Random.seed!(5), gdemo_default, HMC(0.1, 7), MCMCThreads(),
-                            1000, 4)
-            #https://github.com/TuringLang/Turing.jl/issues/1260
-            @test_skip all(chain1.value .== chain2.value)
-            check_gdemo(chain1)
+            @testset "rng" begin
+                model = gdemo_default
+
+                samplers = (HMC(0.1, 7),
+                            PG(10),
+                            IS(),
+                            MH(),
+                            Gibbs(PG(3, :s), HMC(0.4, 8, :m)),
+                            Gibbs(HMC(0.1, 5, :s), ESS(:m)))
+                for sampler in samplers
+                    Random.seed!(5)
+                    chain1 = sample(model, sampler, MCMCThreads(), 1000, 4)
```

Comment:
How threads rng are initialised? Are they sharing the same random seed?

Reply:
That is handled in AbstractMCMC: https://github.com/TuringLang/AbstractMCMC.jl/blob/509b5cf3fe24c6a8809ba6ca98f42ae8165e12e2/src/sample.jl#L215-L216
We sample separate seeds for every chain, using the provided RNG, and create separate copies of the RNG for every thread. When sampling the `i`th chain, the RNG copy of the active thread is seeded with the `i`th seed. Hence the results can be reproduced even with different numbers of threads. The same happens when sampling with multiple processes.

Reply:
thanks for the clarification!